flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -416,13 +426,26 @@ boolean scheduleOrUpdateConsumers(List<List<ExecutionEdge>> consumers) throws Ex
 			final ExecutionState consumerState = consumerVertex.getExecutionState();
 
 			if (consumerState == CREATED) {
-				if (state == RUNNING) {
-					if (!consumerVertex.scheduleForExecution(consumerVertex.getExecutionGraph().getScheduler(), false)) {
-						success = false;
+				consumerVertex.cachePartitionInfo(PartialPartitionInfo.fromEdge(edge));
+
+				future(new Callable<Boolean>(){
+					@Override
+					public Boolean call() throws Exception {
+						try {
+							consumerVertex.scheduleForExecution(
+								consumerVertex.getExecutionGraph().getScheduler(), false);
+						} catch (Exception exception) {
+							fail(new IllegalStateException("Could not schedule consumer " +
+									"vertex " + consumerVertex, exception));
+						}
+
+						return true;
 					}
-				}
-				else {
-					success = false;
+				}, AkkaUtils.globalExecutionContext());
+
+				// double check to resolve race conditions
+				if(consumerVertex.getExecutionState() == RUNNING){
+					consumerVertex.sendPartitionInfos();
```

Comment:
Just to verify: the double check & send relies on the fact that update messages at the task manager are idempotent, right?